[Bug]: Type error in MenuItem.d.ts

### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

MenuItem

### Expected behavior

`import type { Overlay } from '@spectrum-web-components/overlay';` in MenuItem.ts

### Actual behavior

```
node_modules/@spectrum-web-components/menu/src/MenuItem.d.ts:6:30 - error TS2307: Cannot find module 'overlay/src/Overlay.js' or its corresponding type declarations.

6 import type { Overlay } from 'overlay/src/Overlay.js';
                               ~~~~~~~~~~~~~~~~~~~~~~~~

Found 1 error in node_modules/@spectrum-web-components/menu/src/MenuItem.d.ts:6
```
https://github.com/adobe/spectrum-web-components/blob/7fc0c2e143f20ea0aa2dca3470acf7f70e068d11/packages/menu/src/MenuItem.ts#L38

@Westbrook I think this was originally introduced in https://github.com/adobe/spectrum-web-components/commit/5dfb71e5ed26cf8af83ca335a7658938f3f135a6#diff-d6c1bef5a63290802db272d6bf262beb51675b7182fe88d7d93ceda7e32650a9
See MenuItem.ts

### Screenshots

_No response_

### What browsers are you seeing the problem in?

_No response_

### How can we reproduce this issue?

Run tsc on a project that depends on MenuItem
`./node_modules/.bin/tsc --noEmit`

### Sample code that illustrates the problem

_No response_

### Logs taken while reproducing problem

_No response_

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -10,7 +10,7 @@ executors:
 parameters:
     current_golden_images_hash:
         type: string
-        default: a7c50a01ebc6a42895159473849bf1e6f7abebea
+        default: 95fc7baf1ff5ec1894c8692e6674435a0be5a757
     wireit_cache_name:
         type: string
         default: wireit

--- a/packages/menu/src/MenuItem.ts
+++ b/packages/menu/src/MenuItem.ts
@@ -36,7 +36,7 @@ import menuItemStyles from './menu-item.css.js';
 import checkmarkStyles from '@spectrum-web-components/icon/src/spectrum-icon-checkmark.css.js';
 import type { Menu } from './Menu.js';
 import { MutationController } from '@lit-labs/observers/mutation-controller.js';
-import type { Overlay } from 'overlay/src/Overlay.js';
+import type { Overlay } from '@spectrum-web-components/overlay';
 
 /**
  * Duration during which a pointing device can leave an `<sp-menu-item>` element

--- a/packages/overlay/src/topLayerOverTransforms.ts
+++ b/packages/overlay/src/topLayerOverTransforms.ts
@@ -78,10 +78,17 @@ export const topLayerOverTransforms = (): Middleware => ({
                     (containingBlock as unknown as Window)
             ) {
                 const css = getComputedStyle(containingBlock);
+                // The overlay is "over transforms" when the containing block uses specific CSS...
                 overTransforms =
-                    css.transform !== 'none' || css.filter
-                        ? css.filter !== 'none'
-                        : false;
+                    // the `transform` property
+                    css.transform !== 'none' ||
+                    // the `filter` property for anything other than "none"
+                    (css.filter ? css.filter !== 'none' : false) ||
+                    // a value of "paint", "layout", "strict", or "content" for `contain`
+                    ['paint', 'layout', 'strict', 'content'].some((value) =>
+                        (css.contain || '').includes(value)
+                    );
+                // overTransforms = true;
             }
 
             if (onTopLayer && overTransforms && containingBlock) {

--- a/packages/overlay/stories/overlay-element.stories.ts
+++ b/packages/overlay/stories/overlay-element.stories.ts
@@ -152,6 +152,48 @@ longpress.args = {
     type: 'auto',
 };
 
+export const transformed = (args: Properties): TemplateResult => html`
+    <style>
+        .transformed {
+            transform: translateX(-50%);
+            position: absolute;
+            inset: auto;
+            inset-inline-start: 200px;
+            inset-block-start: 200px;
+            inline-size: 100px;
+            block-size: 50px;
+        }
+    </style>
+    <div class="transformed">${Template(args)}</div>
+`;
+transformed.args = {
+    interaction: 'click',
+    placement: 'right',
+    type: 'auto',
+};
+
+export const contained = (args: Properties): TemplateResult => html`
+    <style>
+        .contained {
+            contain: strict;
+            position: absolute;
+            inset: auto;
+            inset-inline-start: 200px;
+            inset-block-start: 200px;
+            inline-size: 200px;
+            block-size: 50px;
+            padding-block: 75px;
+            padding-inline-start: 300px;
+        }
+    </style>
+    <div class="contained">${Template(args)}</div>
+`;
+contained.args = {
+    interaction: 'click',
+    placement: 'right',
+    type: 'auto',
+};
+
 export const all = ({ delayed }: Properties): TemplateResult => html`
     <sp-action-button id="trigger" hold-affordance>
         Open the overlay

